kotlin/display-dimensions/src/main/java/com/esri/arcgisruntime/sample/displaydimensions/MainActivity.kt

```diff
@@ -0,0 +1,140 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.displaydimensions
+
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.constraintlayout.widget.ConstraintLayout
+import com.esri.arcgisruntime.layers.DimensionLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.MobileMapPackage
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.sample.displaydimensions.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.sample.displaydimensions.databinding.DialogLayoutBinding
+import java.io.File
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG: String = MainActivity::class.java.simpleName
+
+    private lateinit var dimensionLayer: DimensionLayer
+    private var isDimensionLayerEnabled: Boolean = true
+    private var isDefinitionEnabled: Boolean = false
+
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val settingsButton: ConstraintLayout by lazy {
+        activityMainBinding.settingsLayout
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // create and load a mobile map package
+        val mobileMapPackage =
+            MobileMapPackage(getExternalFilesDir(null)?.path + "/Edinburgh_Pylon_Dimensions.mmpk")
+
+        mobileMapPackage.addDoneLoadingListener {
+            // check the mmpk has loaded successfully and that it contains a map
+            if (mobileMapPackage.loadStatus == LoadStatus.LOADED && mobileMapPackage.maps.isNotEmpty()) {
+                // add the map from the mobile map package to the map view, and set a min scale to maintain dimension readability
+                mapView.map = mobileMapPackage.maps[0]
+                mapView.map.minScale = 35000.0
+
+                // find the dimension layer within the map
+dimensionLayer =
+                    mapView.map.operationalLayers.firstOrNull { it is DimensionLayer } as DimensionLayer
+            } else {
+                val errorMessage = "Failed to load the mobile map package: " + mobileMapPackage.loadError.message
+                Log.e(TAG, errorMessage)
+                Toast.makeText(
+                    activityMainBinding.root.context,
+                    errorMessage,
+                    Toast.LENGTH_SHORT
+                ).show()
```

Comment:
Is the explicit context useful important? Could inline the whole line with `this`

Toast.makeText(this, errorMessage, Toast.LENGTH_LONG).show()